tests/test_algorithm.py

```diff
@@ -1223,6 +1225,416 @@ def test_position_weights(self):
+                    'auto_close_date': pd.Timestamp('2016-09-07', tz='UTC'),
+                    'exchange': 'CME',
+                    'multiplier': 10,
+                },
+                1007: {
+                    'symbol': 'CLV16',
+                    'root_symbol': 'CL',
+                    'start_date': cls.START_DATE,
+                    'end_date': pd.Timestamp('2016-10-05', tz='UTC'),
+                    'auto_close_date': pd.Timestamp('2016-10-03', tz='UTC'),
+                    'exchange': 'CME',
+                    'multiplier': 10,
+                },
+                1008: {
+                    'symbol': 'CLX16',
+                    'root_symbol': 'CL',
+                    'start_date': cls.START_DATE,
+                    'end_date': cls.END_DATE,
+                    'auto_close_date': cls.END_DATE,
+                    'exchange': 'CME',
+                    'multiplier': 10,
+                },
+            },
+            orient='index',
+        )
+
+    @classmethod
+    def make_future_minute_bar_data(cls):
+        trading_calendar = cls.trading_calendars[Equity]
+
+        minutes = trading_calendar.minutes_for_sessions_in_range(
+            cls.START_DATE, cls.END_DATE,
+        )
+        session_starts = trading_calendar.session_opens_in_range(
+            cls.START_DATE, cls.END_DATE,
+        )
+        frame = pd.DataFrame(
+            {
+                'open': np.NaN,
+                'high': np.NaN,
+                'low': np.NaN,
+                'close': np.NaN,
+                'volume': np.NaN,
+            },
+            index=minutes,
+        )
+
+        # Set all futures contracts to have alternating price values from day
+        # to day, meaning returns will alternate consistently from the same
+        # positive number to the same negative number. This makes it easier to
+        # manually calculate expected shortfall.
+        #
+        # The code for creating an alternating array was found here:
+        # https://stackoverflow.com/a/7154925
+        start_indexes = frame.index.get_indexer(session_starts)
+        prices = np.empty(len(session_starts))
+        prices[::2] = 87
+        prices[1::2] = 100
+        frame.iloc[start_indexes] = np.tile(prices[:, np.newaxis], (1, 5))
+        frame['volume'] = 100
+        frame.fillna(method='ffill', inplace=True)
+
+        return ((sid, frame) for sid in cls.asset_finder.futures_sids)
+
+    def test_expected_shortfall(self):
+        """
+        Test the expected shortfall calculation performed at the end of
+        backtests.
+        """
+        sids = (1, 1004)
+        amounts = (1, 1)
+        calendar = self.trading_calendar
+        af = self.asset_finder
+        equity_1, future_1004 = af.retrieve_all(sids)
+
+        algo = TestPositionWeightsAlgorithm(
+            sids_and_amounts=zip(sids, amounts),
+            sim_params=self.sim_params,
+            env=self.env,
+            benchmark_sid=8554,
```

Comment:
> Is that intentional?

Yes, running in daily makes things easier to reason about and all the expected shortfall calculations are done on daily data so I don't think running in minute mode is necessary, but would be happy to talk more in person.